### ISSUE TITLE
Suggestion: provide a non-model-based widget

### ISSUE THREAD

**Issue by dvarrazzo** [yourlabs/django-autocomplete-light#663]
Hello,

I've just started using DAL in a project of mine: thank you very much for the project. The database is designed independently from django, so it uses some features which are not natural to the framework, e.g. composite primary key. One model I have is an Account object: accounts are in several System and of different AccoutType; in pseudo-sql:

```
System:
    code: text pkey

AccountType:
    id: serial  # Unused, just to make Django happy
    system: text fkey System (code)
    type: text
    primary key (system, text)

Account:
    system: text fkey System (code)
    type: text
    fkey (system, type) AccountType (system, type)
```

Django doesn't handle the multiple pkey/fkey but it's no problem at model level: `Account.type` is just a text field as far as Django is concerned and referential integrity is maintained by the database.

I managed no problem to create a select to populate `Account.type` (the user can select the system first and the type select is populated with the right types for the sytem). However doing this `QuerySetSelectMixin.filter_choices_to_render()` explodes as Django considers the AccountType pkey an integer but the select value is text.

The easy fix was to create a new select type that doesn't assume the object to be a model: after looking at the DAL code I worked out that the following works perfectly:

```python
class Select2(Select2WidgetMixin, WidgetMixin, forms.Select):
    """A Select2 widget with autocompletion not using a primary key."""
    pass
```

I think this object can be useful every time a model needs an autocomplete field (populated from the database or other sources) but the field stored on the model is not the primary key of another model but just some generic value. I find it a very generic use case, but it doesn't seem available in DAL.

Could it be useful? Or am I overlooking something?

**Comment by jpic** [comment]
Hi,

Some work has been started to add this to the docs and example project in #637 and #644. The code changes need a couple of iterations if we keep it, and we'll need tests before we can merge the example. I'm pretty sure @dmosberger would appreciate a helping hand here.

Thanks for contributing to make DAL better !

**Comment by jpic** [comment]
Also, do you think this is a duplicate of #635 ?

**Comment by dvarrazzo** [comment]
I think so: the implementation is pretty much the same :)

I'll close this issue then. Thank you for your work!
